tests/test_target_encoder.py

```diff
@@ -112,3 +112,67 @@ def test_HandleUnknownValue_HaveUnknownInTest_ExpectMean(self):
         obtained = enc.transform(test, test_target)
 
         self.assertEqual(.6, list(obtained['color'])[0])
+
```

Comment:
another nice test would be to check with a "trivial hierarchy", i.e. mapping everything to a unique top level does the same thing as doing target encoding without any hierarchy

Reply:
Yup, agreed, as this currently breaks, I think.

Reply:
Why do you think this breaks? According to the paper this should be fine, right?

Reply:
I just tried it out and it works fine. Here's a complete test. I'd be happy if you include it

```python
    def test_trivial_hierarchy(self):
        trivial_hierarchical_map = {
            'Plant': {
                'Plant': ('Rose', 'Daisy', 'Daffodil', 'Bluebell')
            }
        }

        enc_hier = encoders.TargetEncoder(verbose=1, smoothing=2, min_samples_leaf=2, hierarchy=trivial_hierarchical_map,
                                     cols=['Plant'])
        result_hier = enc_hier.fit_transform(self.hierarchical_cat_example, self.hierarchical_cat_example['target'])
        enc_no_hier = encoders.TargetEncoder(verbose=1, smoothing=2, min_samples_leaf=2, cols=['Plant'])
        result_no_hier = enc_no_hier.fit_transform(self.hierarchical_cat_example, self.hierarchical_cat_example['target'])
        pd.testing.assert_series_equal(result_hier["Plant"], result_no_hier["Plant"])
```